Source/cmake/FindRice.cmake

```
@@ -0,0 +1,139 @@
+# - Try to find LibRice
+#
+# Copyright (C) 2025 Igalia S.L
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1.  Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+# 2.  Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER AND ITS CONTRIBUTORS ``AS
+# IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR ITS
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+# ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#[=======================================================================[.rst:
+FindRice
+--------------
+
+Find Rice headers and libraries.
+
+Imported Targets
+^^^^^^^^^^^^^^^^
+
+``Rice::Proto``
+  The Rice proto library, if found.
+
+``Rice::Io``
+  The Rice io library, if found.
+
+#]=======================================================================]
+
+find_package(PkgConfig QUIET)
+
+if ("Proto" IN_LIST Rice_FIND_COMPONENTS)
+    pkg_check_modules(PC_RICE_PROTO rice-proto)
```

We could set the `Rice_VERSION` variable from the information fetched with pkg-config, and then we can let `find_package_handle_standard_args()` do the version check for us:


```suggestion
    pkg_check_modules(PC_RICE_PROTO rice-proto)
    
    if (NOT Rice_VERSION)
        set(Rice_VERSION ${PC_RICE_PROTO_VERSION})
    endif ()
```